src/components/ViewPCOL/Actions/CancelRequest.tsx

```diff
@@ -10,8 +10,27 @@ import {
   Tooltip,
 } from "@fluentui/react-components";
 import { DeleteIcon } from "@fluentui/react-icons-mdl2";
+import { useParams } from "react-router-dom";
+import { useAddNote } from "src/api/Notes/notesApi";
+import { usePCOL } from "src/api/PCOL/usePCOL";
+import { useStageUpdate } from "src/api/tasks/stage";
+
+const DoneStages = ["Rejected", "Cancelled", "Distributed"];
 
 const CancelRequest = () => {
+  const { program, pcolId } = useParams();
+  const pcol = usePCOL(String(program), Number(pcolId));
+  const addNote = useAddNote(String(program), Number(pcolId));
+  const stageUpdate = useStageUpdate(String(program), Number(pcolId));
```

Comment:
Can't comment in the file directly, since not part of PR -- but line 59 in ApproverDetails.tsx should be updated to add "Cancelled" with "Rejected" so that when Cancelled, there are no tasks marked "Pending" anymore.